lib/ansible/modules/packaging/os/swupd.py

```diff
@@ -0,0 +1,316 @@
+        """Removes a bundle with `swupd bundle-remove bundle`"""
+        if self.module.check_mode:
+            self.module.exit_json(changed=self._is_bundle_installed(bundle))
+
+        if not self._is_bundle_installed(bundle):
+            self.msg = "Bundle %s not installed"
+            return
+
+        cmd = self._get_cmd("bundle-remove %s" % bundle)
+        self._run_cmd(cmd)
+
+        if self.rc == 0:
+            self.changed = True
+            self.msg = "Bundle %s removed" % bundle
+            return
+
+        self.failed = True
+
+    def update_os(self):
+        """Updates the os with `swupd update`"""
+        if self.module.check_mode:
+            self.module.exit_json(changed=self._needs_update())
+
+        if not self._needs_update():
+            self.msg = "There are no updates available"
+            return
+
+        cmd = self._get_cmd("update")
+        self._run_cmd(cmd)
+
+        if self.rc == 0:
+            self.changed = True
+            self.msg = "Update successful"
+            return
+
+        self.failed = True
+
+    def verify_os(self):
+        """Verifies filesystem agains specified or current version"""
+        if self.module.check_mode:
+            self.module.exit_json(changed=self._needs_verify())
+
+        if not self._needs_verify():
+            self.msg = "No files where changed"
+            return
+
+        cmd = self._get_cmd("verify --fix")
+        self._run_cmd(cmd)
+
+        if self.rc == 0 and (self.FILES_REPLACED in self.stdout or self.FILES_FIXED in self.stdout or self.FILES_DELETED in self.stdout):
+            self.changed = True
+            self.msg = "Fix successful"
+            return
+
+        self.failed = True
+
+
+def main():
+    """The main function."""
+    module = AnsibleModule(
+        argument_spec=dict(
+            contenturl=dict(type="str"),
+            format=dict(type="str"),
+            manifest=dict(aliases=["release", "version"], type="int"),
+            name=dict(aliases=["bundle"], type="str"),
+            state=dict(default="present", choices=["present", "absent"], type="str"),
+            update=dict(default=False, type="bool"),
+            url=dict(type="str"),
+            verify=dict(default=False, type="bool"),
+            versionurl=dict(type="str"),
+        ),
+        required_one_of=[["name", "update", "verify"]],
+        mutually_exclusive=[["name", "update", "verify"]],
+        supports_check_mode=True
+    )
+
+    swupd = Swupd(module)
+
+    name = module.params["name"]
+    state = module.params["state"]
+    update = module.params["update"]
+    verify = module.params["verify"]
+
+    if update:
+        swupd.update_os()
+    elif verify:
+        swupd.verify_os()
+    elif state == "present":
+        swupd.install_bundle(name)
+    elif state == "absent":
+        swupd.remove_bundle(name)
+    else:
+        swupd.failed = True
+
+    if swupd.failed:
+        module.fail_json(msg="Unkown Error")
```

Comment:
Typo of "unknown".
Can we give the user any clues, for example asking them to check a log on the remote system?